### ISSUE TITLE
a11y: Fix focus and role on error banner control

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12793]
a11y Issue: 54

When users submit "Login" form empty or with errors, error message get dynamically displayed on the page and user's focus moves to top of the page that resulted in screen reader users finding difficult to find the errors occurred for the form fields.

![image](https://github.com/user-attachments/assets/a9fd9613-4236-4a56-9b11-89a2064a7e3c)

Affects these pages:

Login 
- Invalid username or password. Please try again. 

User: Create
- Password must be at least 12 characters

Import Extension Catalog
-  Unable to determine image name

**Comment by aalves08** [comment]
@rak-phillip given that we've already added the role and focus for the close button, do you agree we can move this one forward? The `user's focus moves to top of the page` isn't something that we can control on form elements... 🤔 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role

<img width="2281" alt="Image" src="https://github.com/user-attachments/assets/136186a8-f875-465f-85f0-890dad39192e" />
<img width="2281" alt="Image" src="https://github.com/user-attachments/assets/59ee8be7-acbb-4b3f-9d57-c5d67c93eee3" />

**Comment by rak-phillip** [comment]
> The user's focus moves to top of the page isn't something that we can control on form elements... 🤔

@aalves08 It looks like this behavior might have something to do with the implementation of `AsyncButton` causing focus to be lost. My bet is that it is something that we are in control of.

**Comment by rak-phillip** [comment]
Cross-posting my comment from the previously raised PR to add more context to this issue https://github.com/rancher/dashboard/pull/13401#issuecomment-2666079830

I think this is an accessibility issue that solely has to deal with the design of the async button and its disabled state. There's two potential way that I can think of addressing this behavior and they both have to deal with modifying the design of async button:

1. Remove the disabled state from the async button. This retains focus on the async button when it enters the "loading" state, but also allows the button to be clicked; we need to ensure that clicks will not perform actions while in the loading state, perhaps also sending a notification that no action is taken because we are waiting for an initial action to complete.
2. Keep the disabled state for the async button and create some hidden input that can receive focus when the button goes into a "loading" state. Only return focus back to the async button if this hidden input still has focus when the async button state shifts to "active" or "error".

There are other approaches that we can take, and I want to emphasize that we need to pay attention to the UX and accessibility of this button. The current behavior has major flaws with regards to maintaining focus.

**Comment by aalves08** [comment]
Great remark and note for future work @rak-phillip 🙇  I did not try to explore the disabled state and it's implications. 

Definitely an avenue worth pursuing. Thanks 🙏

**Comment by gaktive** [comment]
Bigger than size 1 based on first efforts.

**Comment by rak-phillip** [comment]
Moving back to milestone v2.12.0

FYI @nwmac @gaktive

**Comment by izaac** [comment]
Rancher version: `v2.12-c6574346ad86aa68efe1284f63d46d224220009a-head`
UI version: `master 773fbb2`

This issue was validated during the validation of #7607

The focus doesn't go at the top as described in the issue on this Rancher build.
